BSIPA-ModList/UI/WarningUI.cs

```diff
@@ -0,0 +1,149 @@
+        internal static bool firstShow = true;
+
+        public void Awake()
+        {
+            DontDestroyOnLoad(gameObject);
+            SceneManager.activeSceneChanged += this.SceneManager_activeSceneChanged;
+        }
+
+        private void SceneManager_activeSceneChanged(Scene from, Scene to)
+        {
+            if (to.name == "EmptyTransition")
+            {
+                if (Instance != null)
+                {
+                    Instance.StopAllCoroutines();
+                    Destroy(Instance.gameObject);
+                    _mainFlow = null;
+                }
+                Instance = null;
+            }
+        }
+
+        public void Init()
+        {
+            Instance = this;
+
+            Logger.log.Debug("Warning UI Awake");
+            if (firstShow)
+            {
+                firstShow = false;
+                StartCoroutine(LookForUnmetDependencies());
+            }
+        }
+
+        private static MainFlowCoordinator _mainFlow;
+        private static SimpleDialogPromptViewController _warningDialog;
+        private static Queue<WarningEntry> _warningsQueue = new Queue<WarningEntry>();
+
+        private static IEnumerator LookForUnmetDependencies()
+        {
+            Logger.log.Debug("Waiting for MainFlowCoordinator to appear...");
+
+            yield return new WaitWhile(() => FindObjectOfType<MainFlowCoordinator>() == null);
+
+            Logger.log.Debug("Looking for unmet dependencies...");
+
+            lock (Instance)
+            {
+                if (_mainFlow == null)
+                {
+                    _mainFlow = FindObjectOfType<MainFlowCoordinator>();
+                    _warningDialog = _mainFlow.GetPrivateField<SimpleDialogPromptViewController>("_simpleDialogPromptViewController");
+                }
+
+                _warningsQueue.Clear();
+                Dictionary<string, SemVer.Version> loadedPlugins = PluginManager.AllPlugins.Select(x => x.Metadata).Concat(PluginManager.DisabledPlugins).Concat(PluginLoader.ignoredPlugins).ToDictionary(x => x.Id, y => y.Version);
+
+                foreach (var meta in PluginManager.AllPlugins.Select(x => x.Metadata).Concat(PluginManager.DisabledPlugins).Concat(PluginLoader.ignoredPlugins))
+                {
+                    List<string> disabledDependencies = new List<string>();
+                    List<string> missingDependencies = new List<string>();
+                    foreach (var dep in meta.Manifest.Dependencies)
+                    {
+#if DEBUG
+                        Logger.log.Debug($"Looking for dependency {dep.Key} with version range {dep.Value.Intersect(new SemVer.Range("*.*.*"))}");
+#endif
+                        if (loadedPlugins.ContainsKey(dep.Key) && dep.Value.IsSatisfied(loadedPlugins[dep.Key]))
+                            continue;
+
+                        if (loadedPlugins.ContainsKey(dep.Key) && dep.Value.IsSatisfied(loadedPlugins[dep.Key]))
```

Comment:
This looks like the same condition as above? I don't think this block will ever execute.

Reply:
Oops, that's what happens when you copypaste stuff :)